### PR TITLE
Fix GH#16506: add inverted engraving colors for inverted score

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Rebase of #16715

Resolves: #16506

### CODE REVIEW

File: src/framework/draw/types/color.cpp [hunk 1]
```diff
@@ -206,6 +206,14 @@ void Color::setRgba(Rgba rgba)
     m_isValid = true;
 }
 
+Color Color::invert() const
```
Comment (RomanPudashkin) [inline src/framework/draw/types/color.cpp:209]: Please rename it to inverted

File: src/framework/draw/types/color.h [hunk 2]
```diff
@@ -66,6 +66,7 @@ class Color
     void setGreen(int value);
     void setBlue(int value);
     void setAlpha(int value);
+    Color invert() const;
```
Comment (RomanPudashkin) [inline src/framework/draw/types/color.h:69]: Please add an empty line before it

**Review by RomanPudashkin — APPROVED** [review]

File: src/framework/draw/types/color.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply src/framework/draw/types/color.cpp]: done

File: src/framework/draw/types/color.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/framework/draw/types/color.h]: done

**Comment by zacjansheski** [comment]
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approvedhttps://github.com/musescore/MuseScore/issues/16506 FIXED

**Comment by lokman2k5** [comment]
this does not work for guitar tabs! I cannot see my notes! it only works for staff notation, how do I fix this??

**Comment by cbjeukendrup** [comment]
For the record, that's what #11570 is about, but you apparently already found it. It is not trivial to fix, because the real solution would be to make a gap in the staff lines, rather than drawing a box on top of it. It would be slightly better if the box somehow followed the background color, but that's only a workaround. However, rest assured, we have not forgotten about this.

**Comment by lokman2k5** [comment]
I see. I was concerned because the issue has been going on for almost 2 years now. If there is a way I can help that's possible, I know some Cpp, js, etc..
the only workaround I found is using a custom background, still the white box on top looks ugly but that's the only way.

**Comment by borgesguerardidev** [comment]
Really still struggling to export any sort of PNG w/ invisible background. Been trying to work with some bass tablature recently, and the white boxes make it so hard to use.

**Comment by Jojo-Schmitz** [comment]
See #11570

**Comment by borgesguerardidev** [comment]
Sorry, I still don't get how I should fix it!

**Comment by Jojo-Schmitz** [comment]
There is an openb issue, meaning that it isn't fixed yet

**Comment by lokman2k5** [comment]
@borgesguerardidev or anyone struggling with this in tab, see this: https://github.com/musescore/MuseScore/pull/15312

**Comment by Jojo-Schmitz** [comment]
That is closed since 2 years?!

**Pull request merged**